### PR TITLE
[NCL-3881] Introduce new modules for REST Refactor

### PR DESCRIPTION
**Pull request by janinko**
I created new branch **rest-refactor** based from master, so the PRs can be merged into it and the master branch will keep working without breakage from the refactor. This PR is for that branch.

I included also a few example files, just to be sure I did it correctly.

### CODE REVIEW

File: rest/nb-configuration.xml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project-shared-configuration>
+    <!--
+This file contains additional configuration written by modules in the NetBeans IDE.
+The configuration is intended to be shared among all the users of project and
+therefore it is assumed to be part of version control checkout.
+Without this configuration present, some functionality in the IDE may be limited or fail altogether.
+-->
+    <properties xmlns="http://www.netbeans.org/ns/maven-properties-data/1">
```
Comment (jbartece) [inline rest/nb-configuration.xml]: You should ignore such settings and keep it locally IMHO

File: facade/pom.xml [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+
+    <modelVersion>4.0.0</modelVersion>
+
+    <parent>
+        <artifactId>parent</artifactId>
+        <groupId>org.jboss.pnc</groupId>
+        <version>2.0.0-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>facade</artifactId>
```
Comment (jbartece) [inline facade/pom.xml]: Haven't you agreed on a better name with Matej?

File: notifications/pom.xml [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    <description>Websocket notification service.</description>
+  
+  
+    <dependencies>
+      
+        <!-- Project dependencies -->
+        <dependency>
+            <groupId>org.jboss.pnc</groupId>
+            <artifactId>spi</artifactId>
+        </dependency>
```
Comment (jbartece) [inline notifications/pom.xml]: It should communicate using DTOs, so the dependency should be added here.

File: pom.xml [hunk 4]
```diff
@@ -499,6 +541,24 @@
...
+      <dependency>
+        <groupId>io.swagger.core.v3</groupId>
+        <artifactId>swagger-jaxrs2-servlet-initializer</artifactId>
+        <version>2.0.5</version>
+      </dependency>
+
+      <dependency>
+        <groupId>io.swagger.core.v3</groupId>
+        <artifactId>swagger-annotations</artifactId>
+        <version>2.0.5</version>
```
Comment (jbartece) [inline pom.xml]: A variable would be sensible here.

File: facade/pom.xml (hunk 2 above)
Comment (janinko) [reply facade/pom.xml]: No

File: rest/nb-configuration.xml (hunk 1 above)
Comment (janinko) [reply rest/nb-configuration.xml]: Sorry, that was mistake

**Comment by jbartece** [comment]
@janinko The build failed because of missing header. 
[WARNING] Missing header in: /home/jenkins/workspace/PNC_Pull_Requests/constants/pom.xml

**Pull request merged**